File: ZHHDropdownMenu/Classes/ZHHDropdownMenuCell.m
//
//  ZHHDropdownMenuCell.m
//  ZHHDropdownMenu
//
//  Created by 桃色三岁 on 2025/3/24.
//  Copyright © 2025 桃色三岁. All rights reserved.
//

#import "ZHHDropdownMenuCell.h"

@interface ZHHDropdownMenuCell ()
@property (nonatomic, strong, readwrite) UILabel *titleLabel;
@property (nonatomic, strong, readwrite) UIImageView *iconImageView;
@end

@implementation ZHHDropdownMenuCell

- (void)awakeFromNib {
    [super awakeFromNib];
    // Initialization code
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

- (void)prepareForReuse {
    [super prepareForReuse];
    // 重置 cell 状态，避免重用时的显示问题
    self.titleLabel.text = nil;
    self.iconImageView.image = nil;
    self.iconImageView.hidden = NO;
}

- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier{
    if (self = [super initWithStyle:style reuseIdentifier:reuseIdentifier]) {
        self.backgroundColor = UIColor.clearColor;
        self.contentView.backgroundColor = UIColor.clearColor;
        // 允许选中高亮效果
        self.selectionStyle = UITableViewCellSelectionStyleDefault;
        [self setupUI];
    }
    return self;
}

- (void)setupUI {
    [self.contentView addSubview:self.titleLabel];
    [self.contentView addSubview:self.iconImageView];
}

- (UILabel *)titleLabel {
    if (!_titleLabel) {
        _titleLabel = [[UILabel alloc] init];
        _titleLabel.textColor = UIColor.blackColor;
        _titleLabel.font = [UIFont systemFontOfSize:15 weight:UIFontWeightMedium];
        // 移除不必要的 tag，使用属性访问更清晰
        _titleLabel.backgroundColor = [UIColor clearColor];
        _titleLabel.opaque = NO; // 优化渲染性能
    }
    return _titleLabel;
}

- (UIImageView *)iconImageView {
    if (!_iconImageView) {
        _iconImageView = [[UIImageView alloc] init];
        // 移除不必要的 tag，使用属性访问更清晰
        _iconImageView.contentMode = UIViewContentModeScaleAspectFit;
        _iconImageView.backgroundColor = [UIColor clearColor];
        _iconImageView.opaque = NO; // 优化渲染性能
    }
    return _iconImageView;
}
@end

File: ZHHDropdownMenu/Classes/ZHHDropdownMenuCell.swift
//
//  ZHHDropdownMenuCell.swift
//  ZHHDropdownMenu
//
//  Created by 桃色三岁 on 2025/3/24.
//  Copyright © 2025 桃色三岁. All rights reserved.
//

import UIKit

/// 下拉菜单选项 Cell
public class ZHHDropdownMenuCell: UITableViewCell {
    
    // MARK: - Public Properties
    
    /// 标题标签
    public private(set) lazy var titleLabel: UILabel = {
        let label = UILabel()
        label.textColor = .black
        label.font = .systemFont(ofSize: 15, weight: .medium)
        label.backgroundColor = .clear
        label.isOpaque = false // 优化渲染性能
        return label
    }()
    
    /// 图标视图
    public private(set) lazy var iconImageView: UIImageView = {
        let imageView = UIImageView()
        imageView.contentMode = .scaleAspectFit
        imageView.backgroundColor = .clear
        imageView.isOpaque = false // 优化渲染性能
        return imageView
    }()
    
    // MARK: - Initialization
    
    public override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        setupUI()
    }
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
        setupUI()
    }
    
    public override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }
    
    public override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)
        // Configure the view for the selected state
    }
    
    public override func prepareForReuse() {
        super.prepareForReuse()
        // 重置 cell 状态，避免重用时的显示问题
        titleLabel.text = nil
        iconImageView.image = nil
        iconImageView.isHidden = false
    }
    
    // MARK: - Private Methods
    
    private func setupUI() {
        backgroundColor = .clear
        contentView.backgroundColor = .clear
        selectionStyle = .default // 允许选中高亮效果
        
        contentView.addSubview(titleLabel)
        contentView.addSubview(iconImageView)
    }
}

